packages/engine/Source/Scene/Model/ModelRuntimePrimitive.js

```diff
@@ -303,6 +305,10 @@ ModelRuntimePrimitive.prototype.configurePipeline = function (frameState) {
 
   pipelineStages.push(AlphaPipelineStage);
 
+  if (fogRenderable) {
+    pipelineStages.push(FogPipelineStage);
+  }
```

Comment:
I just noticed something: this pipeline stage only needs to be called once per model, not once per primitive, so move this to ModelSceneGraph.